PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -104,40 +104,46 @@ public function connect()
         $errstr = $errno = null;
 
         $remote = sprintf(
-            '%s://%s:%s', 
+            '%s://%s:%s',
             $this->protocol,
             $this->host,
             $this->port
         );
 
         set_error_handler(array($this, 'error_handler'));
-        
-        $this->sock = stream_socket_client(
-            $remote,
-            $errno,
-            $errstr,
-            $this->connection_timeout,
-            STREAM_CLIENT_CONNECT,
-            $this->context
-        );
-        
+
+        try {
+            $this->sock = stream_socket_client(
+                $remote,
+                $errno,
+                $errstr,
+                $this->connection_timeout,
+                STREAM_CLIENT_CONNECT,
+                $this->context
+            );
+        } catch (\Exception $e) {
+            restore_error_handler();
```

Comment:
Would you mind sending a new PR? I think then the try/catch introduced here is also not required